apps/mobile-wallet/src/utils/navigation.ts

```diff
@@ -27,25 +26,12 @@ export const resetNavigation = (
   navigation.reset(getInitialNavigationState(initialRouteName))
 }
 
-export const useAppScreenIsFocused = () => {
-  const navigation = useNavigation()
+export const useScreenIsFocused = useIsFocused
+
+export const useScreenOrModaIsFocused = () => {
   const isAnyModalOpened = useAppSelector(selectIsAnyModalOpened)
   const isTopModal = useIsTopModal()
+  const isScreenFocused = useIsFocused()
```

Comment:
I read some warnings in the docs about this hook, are we sure we need it instead of any other of the proposed solutions?

Reply:
> Note that using this hook triggers a re-render for the component when the screen it's in changes focus. This might cause lags during the animation if your component is heavy. You might want to extract the expensive parts to separate components and use [React.memo](https://react.dev/reference/react/memo) or [React.PureComponent](https://react.dev/reference/react/PureComponent) to minimize re-renders for them.

https://reactnavigation.org/docs/use-is-focused/

Reply:
Let's just roll with it